packages/middleware-retry/src/retryMiddleware.ts

```diff
@@ -31,7 +31,7 @@ export const getRetryPlugin = (
   options: RetryResolvedConfig
 ): Pluggable<any, any> => ({
   applyToStack: clientStack => {
-    if (options.maxRetries > 0) {
+    if (options.maxAttempts > 0) {
```

Comment:
Like my comment above, we only need to retry when `maxAttempts` > 1

Reply:
Fixed in [commit](https://github.com/aws/aws-sdk-js-v3/pull/1244/commits/7d63dea111ba0e0f880ad309406f3165a613b43d)